lib/common.js

```diff
@@ -593,6 +593,22 @@ function deepEqual(expected, actual) {
   return expected === actual
 }
 
+const timeouts = []
+
+function addTimeout(id) {
+  timeouts.push(id)
+}
+
+function removeAllTimeouts() {
+  if (_.isEmpty(timeouts)) {
+    return
+  }
+
+  clearTimeout(timeouts.shift())
+
+  return removeAllTimeouts()
+}
```

Comment:
Instead of recursion this would be better suited for a loop.
```suggestion
function removeAllTimeouts() {
  while(timeouts.length) {
    clearTimeout(timeouts.shift())
  }
}
```

Reply:
i will make the change but would prefer to keep it as shift instead of pop to keep the concept of a queue.